src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -266,147 +232,94 @@ public override Task CloseAsync(WebSocketCloseStatus closeStatus, string? status
             {
                 throw new WebSocketException(WebSocketError.InvalidState, SR.Format(SR.net_WebSockets_InvalidState, fastState, "Connecting, Open, CloseSent, Aborted"));
             }
-
-#if FEATURE_WASM_THREADS
-            return _innerWebSocket!.SynchronizationContext.Post(() =>
+            WebSocketState state;
+            lock (_lockObject)
             {
-                Task promise;
-                lock (_thisLock)
+                state = State;
+                if (state == WebSocketState.Closed)
```

Comment:
Previously `Closed` state was throwing an expcetion, now it's ok, is it intended?